frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+import {
+    lookupTableApi,
+    LookupTable,
+    ValueSet,
+    LookupTables,
+} from "../network/api/LookupTableApi";
+
+export const generateUseLookupTable =
+    <T extends unknown>(tableName: LookupTables) =>
```

Comment:
Is this `unknown` something we can abstract from all possible `T`'s? If not, and if we think it's necessary, a pattern I've seen in the past for type safety of unrelated types is union type checking
```typescript
type ValidExtensions = TypeA | InterfaceOne | TypeC
....<T extends ValidExtensions>
```

Reply:
@kevinhaube turns out we don't need this either! Removed.